lib/quicly.c

```diff
@@ -774,6 +839,64 @@ static int schedule_path_challenge(quicly_conn_t *conn, int is_response, const u
     return 0;
 }
 
+/**
+ * calculate how many CIDs we provide to the peer
+ */
+static uint64_t issued_cid_capacity(const quicly_conn_t *conn)
```

Comment:
Sure. I was actually wondering whether it should be `size_t` or `uint64_t`. I'll go with `size_t` as you suggest.

As of now, I believe `prepare_pending_new_connection_id` is safe because its caller is always checking if `path_id < MAX_PATH_ID` (note: `path_id == MAX_PATH_ID` when `cid_encryptor == NULL`). But anyway I think it's a good idea to guard against future mistakes in multiple layers. I'll add a check in `issued_cid_capacity`.